pos_margin/models/pos_order.py

```diff
@@ -0,0 +1,37 @@
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+import openerp.addons.decimal_precision as dp
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    # Columns Section
+    margin = fields.Float(
+        'Margin', compute='_compute_margin', store=True,
+        digits=dp.get_precision('Product Price'),
+        help="It gives profitability by calculating the difference between"
+        " the Unit Price and the cost price.")
+
+    # Compute Section
+    @api.multi
+    @api.depends('lines.margin')
+    def _compute_margin(self):
+        for order in self:
+            order.margin = sum(order.mapped('lines.margin'))
+
+    @api.multi
+    def recompute_all_margins(self):
+        for line in self.env['pos.order.line'].search([]):
+            line.env.add_todo(line._fields['purchase_price'], line)
+            line.env.add_todo(line._fields['margin'], line)
+            line.recompute()
+
+    @api.multi
+    def recompute_order_margins(self):
+        for order in self.env['pos.order'].search([]):
+            order.env.add_todo(order._fields['margin'], order)
+            order.recompute()
```

Comment:
I don't think that the two buttons are correct. Recompute margin after the initial computation will corrupt data, as the standard price changed, etc...